pilot/pkg/model/push_context.go

```diff
@@ -1674,6 +1674,8 @@ func (ps *PushContext) EnvoyFilters(proxy *Proxy) *EnvoyFilterWrapper {
 				}
 				for _, cp := range cps {
 					if proxyMatch(proxy, cp) {
+						cp.Name = efw.Name
```

Comment:
It feels odd/wrong to modify `cp` (which is part of efw) here. Either its static and we should assign it when we construct the efw, or its dynamic and we cannot safely mutate it?

Reply:
do it in convertToEnvoyFilterWrapper

Reply:
Moved to construction - That is a good idea.